### PR TITLE
Initial release item random option for monster drop

### PR DESCRIPTION
**Pull request by cydh**
* To add item for monster with random option group, see 'db/[pre-]re/mob_drop.txt'
* To add item random option group with random option group, see 'db/[pre-]re/mob_item_randomopt_group.txt'
* Default entries for random options are follow iRO Crimson/Scarlet Weapon based on
  * "[Renewal] Episode 15.2 Crimson Weapon System", see http://www.playragnarok.com/news/updatedetail.aspx?id=280&p=1.
  * File uploaded by Oda (iRO's CM) on https://forums.warpportal.com/index.php?/topic/168763-fansitedb-information-distribution-and-request-station/?p=2495922

Signed-off-by: Cydh Ramdh <cydh@pservero.com>

### CODE REVIEW

File: db/const.txt [hunk 1]
```diff
@@ -2732,3 +2732,5 @@ ARCWANDCLAN	2
 GOLDENMACECLAN	3
 CROSSBOWCLAN	4
 JUMPINGCLAN	5
+
+RDMOPTG_Crimson_Weapon	1
```
Comment (Atemo) [inline db/const.txt]: ```export_constant(RDMOPTG_Crimson_Weapon);``` ?

File: db/re/item_randomopt_group.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_NOTHING,0,0
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_WATER,0,0
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_GROUND,0,0
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_FIRE,0,0
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_WIND,0,0
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_POISON,0,0
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_SAINT,0,0
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_DARKNESS,0,0
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_TELEKINESIS,0,0
+RDMOPTG_Crimson_Weapon,1,RDMOPT_WEAPON_ATTR_UNDEAD,0,0
```
Comment (Atemo) [inline db/re/item_randomopt_group.txt]: Currently charms for crimson weapon don't support ```RDMOPT_WEAPON_ATTR_POISON```, ```RDMOPT_WEAPON_ATTR_TELEKINESIS``` and ```RDMOPT_WEAPON_ATTR_UNDEAD```

Comment (cydh) [inline db/re/item_randomopt_group.txt]: ok

File: db/const.txt (hunk 1 above)
Comment (cydh) [inline db/const.txt]: ok

**Comment by cydh** [comment]
based on the known info,
there are some removed drop from the mobs, I didn't any change about it yet.

**Comment by Badarosk0** [comment]
I do not know if it was just me, but after applying the modification to Skill Steal stopped working. It always fails.
It's working fine?

**Comment by admkakaroto** [comment]
@Badarosk0  i made the same change as you and it worked, but the question remains, can this cause some other problem with this modification?

**Comment by cydh** [comment]
https://github.com/rathena/rathena/pull/1743/files#diff-08420bab8271f3965304fee42cf4b865R5251
must be
```c++
if( md->db->dropitem[i].nameid > 0 && !md->db->dropitem[i].steal_protected && itemdb_exists(md->db->dropitem[i].nameid) && rnd() % 10000 < md->db->dropitem[i].p * rate/100. )
```

**Comment by iubantot** [comment]
cant use rand() for value

**Comment by cydh** [comment]
@iubantot u can't, u can add [these scripts](https://rathena.org/board/topic/108111-for-item-random-option-getitem3-getitembound3-rentitem3-makeitem3-delitem3-countitem3/) then add `OnNPCKillEvent` label for monster u want

**Comment by cydh** [comment]
Closed, renewed to #2065

**Pull request closed without merging**